functions/src/cost.ts

```diff
@@ -41,32 +45,32 @@ export async function calculateCost(dropLoc: GeoPoint) {
   //  - prohibitively expensive to drop a pin right next to an existing one ($10+ in pinnies)
   //  - somewhat expensive to drop a pin that has a few neighbors within a few miles ($0.75 - $1.25)
   //  - cheap to drop a pin several miles away from any other pin ($0.1 - $0.2)
-  return pins.map(d => {
-    const p = d.data();
-
-    const neighborLoc = new GeoPoint(
-      p.location.latitude,
-      p.location.longitude
-    );
+  return pins
+    .map((d) => {
+      const p: Pin = <Pin>d.data();
 
-    const distance = haversineDistance(dropLoc, neighborLoc);
+      const distance = distanceBetween(dropLoc, [
+        p.location.latitude,
+        p.location.longitude,
+      ]);
 
-    const multiplier = ( MAX_COEFF ) / ( (MAX_COEFF - 1) * SCALE * distance + 1 )
-    const cost = multiplier * BASE_COST;
+      const multiplier = MAX_COEFF / ((MAX_COEFF - 1) * SCALE * distance + 1);
+      const cost = multiplier * BASE_COST;
 
-    console.log(`distance: ${distance}, cost: ${cost}`);
+      console.log(`distance: ${distance}, cost: ${cost}`);
 
-    return Math.round(cost);
-  }).reduce((total, n) => total + n, BASE_COST);
+      return Math.round(cost);
+    })
+    .reduce((total, n) => total + n, BASE_COST);
 }
 
 export function calculateReward(pin: Pin) {
   const DECAY_RATE = 0.6;
-  const ogCost = pin.cost ? pin.cost : BASE_COST; // old pins don't have a cost field
-  const reward = pin.finds > 15 // reward multiplier is like 0.01% once we have >= 15 rewards
-    ? BASE_COST                 // so might as well just return the base reward
-    : Math.round(ogCost * Math.pow(DECAY_RATE, pin.finds + 1) + BASE_COST);
+  const reward =
```

Comment:
since we're editing this reward calculation, do we just want to `reward = Math.max(BASE_COST, Math.pow(...))`

Reply:
Nvm I reread your comment and I think that's what you intended.